org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/ClassActionImpl.java

```diff
@@ -1250,6 +1262,9 @@ private int transform(MutableConstantPool constants) throws TransformException {
 							if (outputUtf8 == null) {
 								transformCase = "Direct";
 								outputUtf8 = transformDirectString(inputUtf8);
+                                                                if (outputUtf8 == null) {
+                                                                    outputUtf8 = transformConstantString(inputUtf8, inputName);
```

Comment:
Add a "transformCase" assignment, for example, "Direct per class".

Reply:
Thanks, done.